src/blend.cpp

```diff
@@ -637,40 +651,62 @@ NAN_METHOD(Blend) {
             image->buffer.Reset(buffer.As<v8::Object>());
         } else if (buffer->IsObject()) {
             v8::Local<v8::Object> props = buffer->ToObject();
-            if (props->Has(Nan::New("buffer").ToLocalChecked())) {
-                buffer = props->Get(Nan::New("buffer").ToLocalChecked());
-                if (node::Buffer::HasInstance(buffer)) {
-                    image->buffer.Reset(buffer.As<v8::Object>());
+            if (Nan::New(Image::constructor)->HasInstance(props)) {
+                Image * im = Nan::ObjectWrap::Unwrap<Image>(props);
+                if (im->get()->get_dtype() == mapnik::image_dtype_rgba8) {
+                    image->im_obj = im;
                 }
-            }
-            image->x = props->Get(Nan::New("x").ToLocalChecked())->Int32Value();
-            image->y = props->Get(Nan::New("y").ToLocalChecked())->Int32Value();
-
-            v8::Local<v8::Value> tint_val = props->Get(Nan::New("tint").ToLocalChecked());
-            if (!tint_val.IsEmpty() && tint_val->IsObject()) {
-                v8::Local<v8::Object> tint = tint_val->ToObject();
-                if (!tint.IsEmpty()) {
-                    baton->reencode = true;
-                    std::string msg;
-                    parseTintOps(tint,image->tint,msg);
-                    if (!msg.empty()) {
-                        Nan::ThrowTypeError(msg.c_str());
-                        return;
+            } else {
+                if (props->Has(Nan::New("buffer").ToLocalChecked())) {
+                    buffer = props->Get(Nan::New("buffer").ToLocalChecked());
+                    if (node::Buffer::HasInstance(buffer)) {
+                        image->buffer.Reset(buffer.As<v8::Object>());
+                    } else if (buffer->IsObject()) {
+                        v8::Local<v8::Object> possible_im = buffer.As<v8::Object>();
+                        if (Nan::New(Image::constructor)->HasInstance(possible_im)) {
+                            Image * im = Nan::ObjectWrap::Unwrap<Image>(possible_im);
+                            if (im->get()->get_dtype() == mapnik::image_dtype_rgba8) {
+                                image->im_obj = im;
```

Comment:
Maybe throw if not `mapnik::image_dtype_rgba8` so the user knows what went wrong?

Reply:
I will add a few more errors here, currently it all traps failure below but I think it could be a better message.